### PR TITLE
Add documentation, test

### PR DESCRIPTION
**Pull request by NickBethune**
Add documentation to all methods, add test for check_element_match

### CODE REVIEW

**Comment by justinnhli** [comment]
@crzeller11 Let me know when you want to come by and we can look at this together.

**Comment by crzeller11** [comment]
I can come by in about an hour, after my class ends. Does that work?

Chloe

File: search.py [hunk 1]
```diff
@@ -6,52 +6,82 @@
...
     root = tree.getroot()
     return tree, root
 
 
-def tree_to_list(root):
+def root_to_list(root):
+    """takes in etree root node, parses it to a depth-first ordered list
+    Arguments: root
+    Returns: List of element objects in depth-first order
+    """
```
Comment (justinnhli) [inline search.py]: You need to give the type and a short description of the argument and return values. Keep the heading ("Arguments", "Returns") on separate lines from the description, eg.:

```python

def root_to_list(root):
    """takes in etree root node, parses it to a depth-first ordered list

    Arguments:
        root (FIXME type): FIXME description

    Returns:
        [FIXME type]:  List of element objects in depth-first order
    """

This applies to all other function docstrings you write.

```

File: search.py [hunk 2]
```diff
@@ -6,52 +6,82 @@
...
     ret_list = []
     for element in root.iter():
         ret_list.append(element)
     return ret_list
```
Comment (justinnhli) [inline search.py]: This could be simply:

```python
return list(root.iter())
```

File: search.py [hunk 3]
```diff
@@ -6,52 +6,82 @@
...
-    """return list of all elements of the tag from tree"""
+    """return list of all elements of the tag from tree
+    Arguments: tree [etree]: tree to search
+               tag [string]: tag to search (without namespace)
+    Return: List of elements of type tag
+    """
     search_term = "//mei:" + tag
     r = tree.xpath(search_term,
                    namespaces={'mei': 'http://www.music-encoding.org/ns/mei'})
     return r
```
Comment (justinnhli) [inline search.py]: Consider reformatting as

```python
return tree.xpath(
    "//mei:" + tag,
    namespaces={'mei': 'http://www.music-encoding.org/ns/mei'}
)
```

File: search.py [hunk 4]
```diff
@@ -62,7 +92,10 @@ def get_title(tree):
...
 
 def get_creator(tree):
-    """return a list of all composers (creators) in the piece"""
+    """return a list of all composers (creators) in the piece
+    Arguments: tree [etree]: tree of mei file
+    Return: creators_list [List<element>]: List of elements marking the creator(s) of a piece
+    """
     resp_stmt = get_elements(tree, 'respStmt')
     first = resp_stmt[0]
     arr = first.getchildren()
```
Comment (justinnhli) [inline search.py]: Collapse into a single line:

```python
children = get_elements(tree, 'respStmt')[0].getchildren()
```

File: search.py [hunk 5]
```diff
@@ -136,15 +177,25 @@ def get_mei_from_database(path):
```
Comment (justinnhli) [inline search.py:168]: Condense into list comprehension:

```python
return [filename for filename in os.listdir(path) if file.endswith('.mei')]
```

**Review by justinnhli — CHANGES_REQUESTED** [review]
What you have done is generally good. I added some comments about coding style, but I would actually submit a separate pull request for them. The only other general note is that, as you create the test cases, you should delete the code from the old test() function. I recommend this pull request move all the tests, so please tag me when you're done with that.

**Pull request closed without merging**